packages/terra-compact-interactive-list/src/utils/keyHandlerUtils.js

```diff
@@ -0,0 +1,328 @@
+import getFocusableElements from '../../../terra-table/src/utils/focusManagement';
+
+/**
+ * @param {HTMLElement} element - The element to check if it is a text input
+ * @returns True if the element is a text input. Otherwise, false.
+ */
+export const isTextInput = (element) => {
+  const { tagName } = element;
+  if (tagName.toLowerCase() === 'input') {
+    const validTypes = ['text', 'password', 'number', 'email', 'tel', 'url', 'search', 'date', 'datetime', 'datetime-local', 'time', 'month', 'week'];
+    const inputType = element.type;
+    return validTypes.indexOf(inputType) >= 0;
+  }
+
+  return false;
+};
+
+/**
+ * Move focus to next focusable element outside the container element
+ */
+export const moveFocusFromElement = (containerRef, id, moveForward) => {
+  // get focusable elements from the documen, filter out the container and its children
+  const focusableElements = getFocusableElements(document.body).filter(
+    element => (element.id === id || !containerRef.current.contains(element)),
+  );
+
+  // Identify index of the active element in the DOM excluding data grid children
+  const index = focusableElements.indexOf(containerRef.current);
+  if (index > -1) {
+    // Move focus outside data grid
+    const indexOffset = moveForward ? 1 : -1;
+    let newFocusElement;
+
+    if (index + indexOffset < focusableElements.length) {
+      newFocusElement = focusableElements[index + indexOffset];
+    } else if (focusableElements[0].id !== id) {
+      // eslint-disable-next-line prefer-destructuring
+      newFocusElement = focusableElements[0];
+    }
+
+    if (newFocusElement) {
+      newFocusElement.focus();
+    }
+  }
+};
+
+/**
+ * @param {object} focusedCell - a { row, cell } pair, where row and cell are indexes of currently focused semantic row and a cell within that row.
+ * @param {number} numberOfColumns - a number of visual columns.
+ * @param {boolean} flowHorizontally - sematic rows horizontal flow direction
+ * @param {number} rowsLength - a total number of seamntic rows in the list.
+ * @returns - a { row, cell } indexes pair to focus on.
+ */
+export const handleDownKey = (focusedCell, numberOfColumns, flowHorizontally, rowsLength) => {
```

Comment:
I understand your concern, I'll look into shared logic between these methods and see if it can be improved in my next PR, this one gets too complex to monitor the changes.